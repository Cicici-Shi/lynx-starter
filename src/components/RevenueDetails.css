/* Copyright 2024 The Lynx Authors. All rights reserved. */
/* Licensed under the Apache License Version 2.0 that can be found in the */
/* LICENSE file in the root directory of this source tree. */

/* 收益分配详情组件样式 */
.revenue-details {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1000;
  opacity: 0;
  visibility: hidden;
  transition: opacity 0.3s ease-in-out, visibility 0.3s ease-in-out;
}

.revenue-details.visible {
  opacity: 1;
  visibility: visible;
}

/* 遮罩层 */
.details-overlay {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
}

/* 详情面板 */
.details-panel {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90%;
  max-width: 600px;
  max-height: 80vh;
  background: var(--rp-panel, #fcfcfc);
  border-radius: 16px;
  box-shadow: var(--rp-shadow, 0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04));
  overflow: hidden;
  animation: slideIn 0.3s ease-out;
}

@keyframes slideIn {
  from {
    opacity: 0;
    transform: translate(-50%, -50%) scale(0.95);
  }
  to {
    opacity: 1;
    transform: translate(-50%, -50%) scale(1);
  }
}

/* 头部 */
.details-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 20px 24px;
  border-bottom: 1px solid var(--rp-border, #e5e7eb);
  background: var(--rp-bg, #f8f7f6);
}

.details-title {
  font-size: 18px;
  font-weight: 600;
  color: var(--rp-text, #0f172a);
  margin: 0;
}

.header-actions {
  display: flex;
  align-items: center;
  gap: 16px;
}

/* 关闭按钮 */
.close-button {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  border-radius: 8px;
  cursor: pointer;
  transition: background-color 0.15s ease;
}

.close-button:hover {
  background: var(--rp-border, #e5e7eb);
}

.close-icon {
  color: var(--rp-muted, #6b7280);
}

/* 内容区域 */
.details-content {
  max-height: calc(80vh - 80px);
  overflow-y: auto;
}

/* 基本信息 */
.basic-info {
  padding: 20px 24px;
  border-bottom: 1px solid var(--rp-border, #f3f4f6);
}

.info-item {
  display: flex;
  align-items: flex-start;
  margin-bottom: 12px;
}

.info-item:last-child {
  margin-bottom: 0;
}

.info-label {
  font-size: 14px;
  color: var(--rp-muted, #6b7280);
  font-weight: 500;
  min-width: 80px;
  margin-right: 12px;
}

.info-value {
  font-size: 14px;
  color: var(--rp-text, #0f172a);
  flex: 1;
}

.total-amount {
  font-weight: 600;
  color: var(--rp-brand, #10b981);
  font-size: 16px;
}

.calculation-basis {
  align-items: flex-start;
}

.basis-text {
  line-height: 1.5;
}

/* 分配详情区域 */
.distribution-section {
  padding: 20px 24px;
  border-bottom: 1px solid var(--rp-border, #f3f4f6);
}

.section-title {
  font-size: 16px;
  font-weight: 600;
  color: var(--rp-text, #0f172a);
  margin-bottom: 16px;
  display: block;
}

.distribution-list {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.distribution-item {
  padding: 16px;
  background: var(--rp-bg, #f8f7f6);
  border-radius: 12px;
  border: 1px solid var(--rp-border, #f3f4f6);
}

.item-header {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 8px;
}

.party-info {
  display: flex;
  align-items: center;
  gap: 8px;
  flex: 1;
}

.category-indicator {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  flex-shrink: 0;
}

.party-name {
  font-size: 14px;
  font-weight: 600;
  color: var(--rp-text, #0f172a);
}

.party-category {
  font-size: 12px;
  color: var(--rp-muted, #6b7280);
}

.party-amount {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 2px;
}

.percentage {
  font-size: 16px;
  font-weight: 700;
  color: var(--rp-text, #0f172a);
}

.amount {
  font-size: 12px;
  color: var(--rp-muted, #6b7280);
}

.distribution-reason {
  font-size: 13px;
  color: var(--rp-muted, #6b7280);
  line-height: 1.4;
  margin-bottom: 12px;
}

/* 百分比可视化条 */
.percentage-bar {
  height: 4px;
  background: var(--rp-border, #e5e7eb);
  border-radius: 2px;
  overflow: hidden;
}

.percentage-fill {
  height: 100%;
  border-radius: 2px;
  transition: width 0.3s ease;
  animation: barGrow 0.6s ease-out;
}

@keyframes barGrow {
  from {
    width: 0%;
  }
}

/* 分配汇总 */
.distribution-summary {
  padding: 20px 24px;
  background: var(--rp-bg, #f8f7f6);
  display: flex;
  justify-content: space-between;
  gap: 20px;
}

.summary-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 4px;
}

.summary-label {
  font-size: 12px;
  color: var(--rp-muted, #6b7280);
  font-weight: 500;
}

.summary-value {
  font-size: 16px;
  font-weight: 700;
  color: var(--rp-text, #0f172a);
}

/* 响应式设计 */
@media (max-width: 640px) {
  .details-panel {
    width: 95%;
    max-height: 90vh;
  }

  .details-header {
    padding: 16px 20px;
  }

  .details-title {
    font-size: 16px;
  }

  .basic-info,
  .distribution-section {
    padding: 16px 20px;
  }

  .distribution-summary {
    padding: 16px 20px;
    flex-direction: column;
    gap: 12px;
  }

  .summary-item {
    flex-direction: row;
    justify-content: space-between;
  }
}

/* 动画优化 */
@media (prefers-reduced-motion: reduce) {
  .revenue-details,
  .details-panel,
  .close-button,
  .percentage-fill {
    transition: none;
    animation: none;
  }
}