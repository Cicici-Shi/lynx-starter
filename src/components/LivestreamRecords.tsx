// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

import { Icon } from "./Icon.tsx";
import { RevenueDetails } from "./RevenueDetails.tsx";
import type {
  LivestreamRecord,
  RevenueDistribution,
} from "../types/chatbot.ts";

/**
 * 直播记录展示组件属性接口
 */
export interface LivestreamRecordsProps {
  records: LivestreamRecord[];
  expandedRecordId: string | null;
  revenueDistributions: Map<string, RevenueDistribution>;
  onRecordClick: (recordId: string) => void;
  onRecordCollapse: () => void;
}

/**
 * 直播记录展示组件
 *
 * 功能特性:
 * - 展示最近3-5条直播记录
 * - 支持点击展开/收起收益分配详情
 * - 响应式卡片布局
 * - 状态指示器(完成/直播中/已安排)
 * - 空状态展示
 */
export function LivestreamRecords({
  records,
  expandedRecordId,
  revenueDistributions,
  onRecordClick,
  onRecordCollapse,
}: LivestreamRecordsProps) {
  // 如果没有记录，显示空状态
  if (records.length === 0) {
    return (
      <view className="records-section">
        <view className="section-header">
          <Icon name="livestream" size={20} className="section-icon" />
          <text className="section-title">最近直播</text>
        </view>

        <view className="empty-state">
          <Icon name="empty" size={48} className="empty-icon" />
          <text className="empty-title">暂无直播记录</text>
          <text className="empty-description">
            还没有直播记录，开始你的第一场直播吧！
          </text>
        </view>
      </view>
    );
  }

  return (
    <view className="records-section">
      <view className="section-header">
        <Icon name="livestream" size={20} className="section-icon" />
        <text className="section-title">最近直播</text>
      </view>

      <scroll-view
        className="records-scroll"
        scroll-orientation="vertical"
        scroll-bar-enable={true}
      >
        <view className="records-list">
          {records.map((record) => (
            <RecordCard
              key={record.id}
              record={record}
              isExpanded={expandedRecordId === record.id}
              revenueDistribution={revenueDistributions.get(record.id)}
              onRecordClick={onRecordClick}
              onRecordCollapse={onRecordCollapse}
            />
          ))}
        </view>
      </scroll-view>
    </view>
  );
}

/**
 * 单条直播记录卡片组件属性接口
 */
export interface RecordCardProps {
  record: LivestreamRecord;
  isExpanded: boolean;
  revenueDistribution?: RevenueDistribution;
  onRecordClick: (recordId: string) => void;
  onRecordCollapse: () => void;
}

/**
 * 单条直播记录卡片组件
 *
 * 功能特性:
 * - 显示直播基本信息（标题、日期、时长、收益）
 * - 状态指示器显示直播状态
 * - 点击展开/收起收益分配详情
 * - 收益分配详情可视化展示
 */
export function RecordCard({
  record,
  isExpanded,
  revenueDistribution,
  onRecordClick,
  onRecordCollapse,
}: RecordCardProps) {
  /**
   * 格式化日期显示
   */
  const formatDate = (dateString: string): string => {
    const date = new Date(dateString);
    return date.toLocaleDateString("zh-CN", {
      year: "numeric",
      month: "2-digit",
      day: "2-digit",
    });
  };

  /**
   * 格式化金额显示
   */
  const formatAmount = (amount: number): string => {
    return amount.toLocaleString("zh-CN", {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    });
  };

  /**
   * 获取状态显示文本
   */
  const getStatusText = (status: LivestreamRecord["status"]): string => {
    switch (status) {
      case "completed":
        return "已完成";
      case "live":
        return "直播中";
      case "scheduled":
        return "已安排";
      default:
        return "未知";
    }
  };

  return (
    <view className="record-item">
      {/* 直播记录卡片主体 */}
      <view
        className={`record-card ${isExpanded ? "expanded" : ""}`}
        bindtap={() => onRecordClick(record.id)}
      >
        <view className="record-main">
          <view className="record-info">
            <text className="record-title">{record.title}</text>
            <view className="record-meta">
              <text className="record-date">{formatDate(record.date)}</text>
              <text className="record-duration">{record.duration}</text>
              <text className="record-revenue">
                ¥{formatAmount(record.totalRevenue)}
              </text>
            </view>
            {/* 额外信息：观看人数 */}
            {record.viewerCount && (
              <view className="record-stats">
                <text className="stats-item">
                  观看: {record.viewerCount.toLocaleString()}人
                </text>
                {record.peakViewers && (
                  <text className="stats-item">
                    峰值: {record.peakViewers.toLocaleString()}人
                  </text>
                )}
              </view>
            )}
          </view>

          <view className="record-status">
            <view className="status-indicator">
              <view className={`status-dot status-${record.status}`}></view>
              <text className="status-text">
                {getStatusText(record.status)}
              </text>
            </view>
            <Icon
              name={isExpanded ? "dropup" : "dropdown"}
              size={16}
              className="expand-icon"
            />
          </view>
        </view>
      </view>

      {/* 收益分配详情展开面板 */}
      {revenueDistribution && (
        <RevenueDetails
          distribution={revenueDistribution}
          isVisible={isExpanded}
          onClose={onRecordCollapse}
        />
      )}
    </view>
  );
}
