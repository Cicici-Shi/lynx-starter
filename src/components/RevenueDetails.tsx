// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

import { Icon } from "./Icon.tsx";
import type { RevenueDistribution } from "../types/chatbot.ts";
import type { RevenueDetailsProps } from "../types/components.ts";
import "./RevenueDetails.css";

/**
 * 获取参与方类别的显示颜色
 */
function getCategoryColor(category: string, index: number): string {
  const colors = {
    streamer: "#10b981", // 绿色
    platform: "#3b82f6", // 蓝色
    partner: "#f59e0b", // 橙色
    other: "#6b7280", // 灰色
  };

  // 如果类别不存在，使用备用颜色数组
  const fallbackColors = [
    "#10b981",
    "#3b82f6",
    "#f59e0b",
    "#ef4444",
    "#8b5cf6",
    "#06b6d4",
  ];

  return (
    colors[category as keyof typeof colors] ||
    fallbackColors[index % fallbackColors.length]
  );
}

/**
 * 收益分配详情组件
 *
 * 功能特性:
 * - 展示分配详情和分配比例、金额和理由
 * - 实现平滑的展开/收起动画效果
 * - 显示分配比例可视化条
 */
export function RevenueDetails({
  distribution,
  isVisible,
  onClose,
}: RevenueDetailsProps) {
  /**
   * 格式化金额显示
   */
  const formatAmount = (amount: number): string => {
    return amount.toLocaleString("zh-CN", {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    });
  };

  /**
   * 格式化计算时间
   */
  const formatCalculatedTime = (dateString: string): string => {
    const date = new Date(dateString);
    return date.toLocaleString("zh-CN", {
      year: "numeric",
      month: "2-digit",
      day: "2-digit",
      hour: "2-digit",
      minute: "2-digit",
    });
  };

  if (!isVisible) {
    return null;
  }

  return (
    <view className={`revenue-details ${isVisible ? "visible" : ""}`}>
      <view className="details-overlay" bindtap={onClose}></view>
      <view className="details-panel">
        {/* 头部 */}
        <view className="details-header">
          <text className="details-title">收益分配详情</text>
          <view className="header-actions">
            <view className="close-button" bindtap={onClose}>
              <Icon name="close" size={20} className="close-icon" />
            </view>
          </view>
        </view>

        <scroll-view
          className="details-content"
          scroll-orientation="vertical"
          scroll-bar-enable={true}
        >
          {/* 基本信息 */}
          <view className="basic-info">
            <view className="info-item">
              <text className="info-label">总收益：</text>
              <text className="info-value total-amount">
                {distribution.currency} {formatAmount(distribution.totalAmount)}
              </text>
            </view>
            <view className="info-item">
              <text className="info-label">计算时间：</text>
              <text className="info-value">
                {formatCalculatedTime(distribution.calculatedAt)}
              </text>
            </view>
            <view className="info-item calculation-basis">
              <text className="info-label">计算依据：</text>
              <text className="info-value basis-text">
                {distribution.calculationBasis}
              </text>
            </view>
          </view>

          {/* 详细分配列表 */}
          <view className="distribution-section">
            <text className="section-title">分配详情</text>
            <view className="distribution-list">
              {distribution.distributions.map((item, index) => (
                <view
                  key={`${item.party}-${index}`}
                  className="distribution-item"
                >
                  <view className="item-header">
                    <view className="party-info">
                      <view
                        className="category-indicator"
                        style={{
                          backgroundColor: getCategoryColor(
                            item.category,
                            index,
                          ),
                        }}
                      ></view>
                      <text className="party-name">{item.party}</text>
                      <text className="party-category">({item.category})</text>
                    </view>
                    <view className="party-amount">
                      <text className="percentage">{item.percentage}%</text>
                      <text className="amount">
                        {distribution.currency} {formatAmount(item.amount)}
                      </text>
                    </view>
                  </view>

                  {/* 分配理由 */}
                  <text className="distribution-reason">{item.reason}</text>

                  {/* 分配比例可视化条 */}
                  <view className="percentage-bar">
                    <view
                      className="percentage-fill"
                      style={{
                        width: `${item.percentage}%`,
                        backgroundColor: getCategoryColor(item.category, index),
                      }}
                    ></view>
                  </view>
                </view>
              ))}
            </view>
          </view>

          {/* 分配汇总 */}
          <view className="distribution-summary">
            <view className="summary-item">
              <text className="summary-label">总计比例：</text>
              <text className="summary-value">
                {distribution.distributions.reduce(
                  (sum, item) => sum + item.percentage,
                  0,
                )}
                %
              </text>
            </view>
            <view className="summary-item">
              <text className="summary-label">总计金额：</text>
              <text className="summary-value">
                {distribution.currency}{" "}
                {formatAmount(
                  distribution.distributions.reduce(
                    (sum, item) => sum + item.amount,
                    0,
                  ),
                )}
              </text>
            </view>
          </view>
        </scroll-view>
      </view>
    </view>
  );
}
