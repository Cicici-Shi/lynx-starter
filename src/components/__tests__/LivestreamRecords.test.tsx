// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

import "@testing-library/jest-dom";
import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, getQueriesForElement } from "@lynx-js/react/testing-library";
import { LivestreamRecords, RecordCard } from "../LivestreamRecords.tsx";
import { RevenueDetails } from "../RevenueDetails.tsx";
import type {
  LivestreamRecord,
  RevenueDistribution,
} from "../../types/chatbot.ts";

// Mock Icon component
vi.mock("../Icon.tsx", () => ({
  Icon: ({
    name,
    size,
    className,
  }: {
    name: string;
    size: number;
    className?: string;
  }) => (
    <view className={`icon ${className}`} data-name={name} data-size={size}>
      {name}
    </view>
  ),
}));

describe("LivestreamRecords 组件测试", () => {
  const mockRecords: LivestreamRecord[] = [
    {
      id: "1",
      title: "测试直播1",
      date: "2024-12-28T20:00:00Z",
      duration: "2h 30m",
      totalRevenue: 15680.5,
      status: "completed",
      viewerCount: 1250,
      peakViewers: 1580,
    },
    {
      id: "2",
      title: "测试直播2",
      date: "2024-12-25T19:30:00Z",
      duration: "1h 45m",
      totalRevenue: 8920.3,
      status: "live",
      viewerCount: 890,
      peakViewers: 1120,
    },
  ];

  const mockDistributions = new Map<string, RevenueDistribution>([
    [
      "1",
      {
        recordId: "1",
        totalAmount: 15680.5,
        currency: "CNY",
        calculationBasis: "基于观看时长、互动率和打赏金额的综合算法",
        calculatedAt: "2024-12-28T22:30:00Z",
        distributions: [
          {
            party: "主播",
            percentage: 60,
            amount: 9408.3,
            reason: "主要内容创作者",
            category: "streamer",
          },
          {
            party: "平台",
            percentage: 40,
            amount: 6272.2,
            reason: "提供技术支持",
            category: "platform",
          },
        ],
      },
    ],
  ]);

  const mockProps = {
    records: mockRecords,
    expandedRecordId: null,
    revenueDistributions: mockDistributions,
    onRecordClick: vi.fn(),
    onRecordCollapse: vi.fn(),
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("应该正确渲染组件结构", () => {
    render(<LivestreamRecords {...mockProps} />);

    // 检查组件是否正确渲染
    expect(elementTree.root).toMatchSnapshot();
  });

  it("应该显示正确的记录数量", () => {
    render(<LivestreamRecords {...mockProps} />);

    // 检查是否有正确数量的记录项
    const recordItems = elementTree.root?.querySelectorAll(".record-item");
    expect(recordItems).toHaveLength(2);
  });

  it("应该在空状态下显示空提示", () => {
    const emptyProps = {
      ...mockProps,
      records: [],
    };

    render(<LivestreamRecords {...emptyProps} />);

    // 检查是否有空状态容器
    const emptyState = elementTree.root?.querySelector(".empty-state");
    expect(emptyState).toBeTruthy();
  });

  it("应该在展开状态下显示收益详情", () => {
    const expandedProps = {
      ...mockProps,
      expandedRecordId: "1",
    };

    render(<LivestreamRecords {...expandedProps} />);

    // 检查是否有收益详情容器
    const revenueDetails = elementTree.root?.querySelector(".revenue-details");
    expect(revenueDetails).toBeTruthy();
  });
});

describe("RecordCard 组件测试", () => {
  const mockRecord: LivestreamRecord = {
    id: "1",
    title: "测试直播",
    date: "2024-12-28T20:00:00Z",
    duration: "2h 30m",
    totalRevenue: 15680.5,
    status: "completed",
    viewerCount: 1250,
    peakViewers: 1580,
  };

  const mockDistribution: RevenueDistribution = {
    recordId: "1",
    totalAmount: 15680.5,
    currency: "CNY",
    calculationBasis: "基于观看时长、互动率和打赏金额的综合算法",
    calculatedAt: "2024-12-28T22:30:00Z",
    distributions: [
      {
        party: "主播",
        percentage: 60,
        amount: 9408.3,
        reason: "主要内容创作者",
        category: "streamer",
      },
    ],
  };

  const mockProps = {
    record: mockRecord,
    isExpanded: false,
    revenueDistribution: mockDistribution,
    onRecordClick: vi.fn(),
    onRecordCollapse: vi.fn(),
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("应该正确渲染记录卡片结构", () => {
    render(<RecordCard {...mockProps} />);

    // 检查基本结构
    const recordCard = elementTree.root?.querySelector(".record-card");
    expect(recordCard).toBeTruthy();

    const recordTitle = elementTree.root?.querySelector(".record-title");
    expect(recordTitle).toBeTruthy();

    const recordMeta = elementTree.root?.querySelector(".record-meta");
    expect(recordMeta).toBeTruthy();
  });

  it("应该在展开状态下显示收益详情", () => {
    const expandedProps = { ...mockProps, isExpanded: true };

    render(<RecordCard {...expandedProps} />);

    // 检查是否有收益详情
    const revenueDetails = elementTree.root?.querySelector(".revenue-details");
    expect(revenueDetails).toBeTruthy();
  });

  it("应该正确显示不同的直播状态", () => {
    const liveRecord = { ...mockRecord, status: "live" as const };
    const liveProps = { ...mockProps, record: liveRecord };

    render(<RecordCard {...liveProps} />);

    // 检查状态点样式
    const statusDot = elementTree.root?.querySelector(
      ".status-dot.status-live",
    );
    expect(statusDot).toBeTruthy();
  });
});

describe("RevenueDetails 组件测试", () => {
  const mockDistribution: RevenueDistribution = {
    recordId: "1",
    totalAmount: 15680.5,
    currency: "CNY",
    calculationBasis: "基于观看时长、互动率和打赏金额的综合算法",
    calculatedAt: "2024-12-28T22:30:00Z",
    distributions: [
      {
        party: "主播",
        percentage: 60,
        amount: 9408.3,
        reason: "主要内容创作者，承担直播策划和执行",
        category: "streamer",
      },
      {
        party: "平台",
        percentage: 40,
        amount: 6272.2,
        reason: "提供技术支持、流量推广和平台服务",
        category: "platform",
      },
    ],
  };

  const mockProps = {
    distribution: mockDistribution,
    isVisible: true,
    onClose: vi.fn(),
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("应该正确渲染收益分配详情结构", () => {
    render(<RevenueDetails {...mockProps} />);

    // 检查主要结构元素
    const detailsHeader = elementTree.root?.querySelector(".details-header");
    expect(detailsHeader).toBeTruthy();

    const detailsContent = elementTree.root?.querySelector(".details-content");
    expect(detailsContent).toBeTruthy();

    const calculationBasis =
      elementTree.root?.querySelector(".calculation-basis");
    expect(calculationBasis).toBeTruthy();

    const distributionList =
      elementTree.root?.querySelector(".distribution-list");
    expect(distributionList).toBeTruthy();
  });

  it("应该显示正确数量的分配项目", () => {
    render(<RevenueDetails {...mockProps} />);

    // 检查分配项目数量
    const distributionItems =
      elementTree.root?.querySelectorAll(".distribution-item");
    expect(distributionItems).toHaveLength(2);
  });

  it("应该显示分配比例可视化条", () => {
    render(<RevenueDetails {...mockProps} />);

    // 检查百分比条
    const percentageBars =
      elementTree.root?.querySelectorAll(".percentage-bar");
    expect(percentageBars).toHaveLength(2);

    const percentageFills =
      elementTree.root?.querySelectorAll(".percentage-fill");
    expect(percentageFills).toHaveLength(2);
  });

  it("应该显示分类指示器", () => {
    render(<RevenueDetails {...mockProps} />);

    // 检查分类指示器
    const categoryIndicators = elementTree.root?.querySelectorAll(
      ".category-indicator",
    );
    expect(categoryIndicators).toHaveLength(2);
  });
});

describe("组件功能测试", () => {
  it("应该正确处理格式化函数", () => {
    const mockRecord: LivestreamRecord = {
      id: "1",
      title: "测试直播",
      date: "2024-12-28T20:00:00Z",
      duration: "2h 30m",
      totalRevenue: 1234.56,
      status: "completed",
    };

    const mockProps = {
      record: mockRecord,
      isExpanded: false,
      onRecordClick: vi.fn(),
      onRecordCollapse: vi.fn(),
    };

    render(<RecordCard {...mockProps} />);

    // 检查组件是否正确渲染（不依赖具体文本内容）
    const recordCard = elementTree.root?.querySelector(".record-card");
    expect(recordCard).toBeTruthy();
  });

  it("应该正确处理不同状态的样式", () => {
    const statuses: Array<LivestreamRecord["status"]> = [
      "completed",
      "live",
      "scheduled",
    ];

    statuses.forEach((status) => {
      const mockRecord: LivestreamRecord = {
        id: "1",
        title: "测试直播",
        date: "2024-12-28T20:00:00Z",
        duration: "2h 30m",
        totalRevenue: 1000,
        status,
      };

      const mockProps = {
        record: mockRecord,
        isExpanded: false,
        onRecordClick: vi.fn(),
        onRecordCollapse: vi.fn(),
      };

      render(<RecordCard {...mockProps} />);

      // 检查状态点是否有正确的类名
      const statusDot = elementTree.root?.querySelector(
        `.status-dot.status-${status}`,
      );
      expect(statusDot).toBeTruthy();
    });
  });
});
