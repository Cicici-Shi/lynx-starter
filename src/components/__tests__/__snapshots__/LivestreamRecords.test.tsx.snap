// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LivestreamRecords 组件测试 > 应该正确渲染组件结构 1`] = `
<page>
  <view
    class="records-section"
  >
    <view
      class="section-header"
    >
      <wrapper>
        <view
          class="icon section-icon"
          data-name="livestream"
          data-size="20"
        >
          livestream
        </view>
      </wrapper>
      <text
        class="section-title"
      >
        最近直播
      </text>
    </view>
    <scroll-view
      class="records-scroll"
      scroll-bar-enable="true"
      scroll-orientation="vertical"
    >
      <view
        class="records-list"
      >
        <view
          class="record-item"
        >
          <view
            class="record-card "
          >
            <view
              class="record-main"
            >
              <view
                class="record-info"
              >
                <text
                  class="record-title"
                >
                  测试直播1
                </text>
                <view
                  class="record-meta"
                >
                  <text
                    class="record-date"
                  >
                    2024/12/29
                  </text>
                  <text
                    class="record-duration"
                  >
                    2h 30m
                  </text>
                  <text
                    class="record-revenue"
                  >
                    ¥
                    <wrapper>
                      15,680.50
                    </wrapper>
                  </text>
                </view>
                <wrapper>
                  <view
                    class="record-stats"
                  >
                    <text
                      class="stats-item"
                    >
                      观看: 
                      <wrapper>
                        1,250
                      </wrapper>
                      人
                    </text>
                    <wrapper>
                      <text
                        class="stats-item"
                      >
                        峰值: 
                        <wrapper>
                          1,580
                        </wrapper>
                        人
                      </text>
                    </wrapper>
                  </view>
                </wrapper>
              </view>
              <view
                class="record-status"
              >
                <view
                  class="status-indicator"
                >
                  <view
                    class="status-dot status-completed"
                  />
                  <text
                    class="status-text"
                  >
                    已完成
                  </text>
                </view>
                <wrapper>
                  <view
                    class="icon expand-icon"
                    data-name="dropdown"
                    data-size="16"
                  >
                    dropdown
                  </view>
                </wrapper>
              </view>
            </view>
          </view>
          <wrapper />
        </view>
        <view
          class="record-item"
        >
          <view
            class="record-card "
          >
            <view
              class="record-main"
            >
              <view
                class="record-info"
              >
                <text
                  class="record-title"
                >
                  测试直播2
                </text>
                <view
                  class="record-meta"
                >
                  <text
                    class="record-date"
                  >
                    2024/12/26
                  </text>
                  <text
                    class="record-duration"
                  >
                    1h 45m
                  </text>
                  <text
                    class="record-revenue"
                  >
                    ¥
                    <wrapper>
                      8,920.30
                    </wrapper>
                  </text>
                </view>
                <wrapper>
                  <view
                    class="record-stats"
                  >
                    <text
                      class="stats-item"
                    >
                      观看: 
                      <wrapper>
                        890
                      </wrapper>
                      人
                    </text>
                    <wrapper>
                      <text
                        class="stats-item"
                      >
                        峰值: 
                        <wrapper>
                          1,120
                        </wrapper>
                        人
                      </text>
                    </wrapper>
                  </view>
                </wrapper>
              </view>
              <view
                class="record-status"
              >
                <view
                  class="status-indicator"
                >
                  <view
                    class="status-dot status-live"
                  />
                  <text
                    class="status-text"
                  >
                    直播中
                  </text>
                </view>
                <wrapper>
                  <view
                    class="icon expand-icon"
                    data-name="dropdown"
                    data-size="16"
                  >
                    dropdown
                  </view>
                </wrapper>
              </view>
            </view>
          </view>
          <wrapper />
        </view>
      </view>
    </scroll-view>
  </view>
</page>
`;
